### PR TITLE
Token matcher

### PR DESCRIPTION
**Pull request by vonschau**
Now it’s possible to match page against URL with tokens ({token}).

### CODE REVIEW

File: Entity/PageManager.php [hunk 1]
```diff
@@ -118,6 +118,40 @@ public function findOneByUrl($site, $url)
...
+     *
+     * @param Page $page
+     * @param $url
+     * @return bool
+     */
+    protected function matches(Page $page, $url)
+    {
+        $purl = $page->getUrl();
+        $pattern = '#^'.$purl.'$#';
+        preg_match_all('/{[a-z]+}/', $pattern, $matches);
```
Comment (jbazant) [inline Entity/PageManager.php]: Why $pattern and not $purl?

File: Entity/PageManager.php [hunk 2]
```diff
@@ -118,6 +118,40 @@ public function findOneByUrl($site, $url)
...
+
+        if ($tokens = $matches[0]) {
```
Comment (jbazant) [inline Entity/PageManager.php]: Unnecessary test. $matches should be always an array. (Side note: Assignment as only operation of condition is considered as bad practice by majority of code standards.)

**Comment by jbazant** [comment]
Add some fixes to Entity/PageManager.php file (lines 132 and 134).

I am not sure whether line 57 of file Route/CmsPageRouter.php works as expected, because $page->parameters is array of arrays. Pls recheck it.

Rest of the patch is OK.

**Comment by vonschau** [comment]
There's array_merge on line 57. I think it's not array of arrays.

**Comment by vonschau** [comment]
OK

**Pull request merged**